bench/irmin-pack/trace_replay.ml

```diff
@@ -367,42 +371,29 @@ module Make (Store : Store) = struct
             `Replay
               {
                 path_conversion = config.path_conversion;
-                artefacts_dir = config.artefacts_dir;
+                artefacts_dir = config.path_to_artefacts;
               };
           inode_config = (entries, entries, stable_hash);
           store_type = config.store_type;
         }
     in
-    let stats = Stat_collector.create_file stat_path c config.store_dir in
+    let stats = Stat_collector.create_file stat_path c root in
     Irmin_pack.Stats.reset_stats ();
-    let+ summary_opt =
-      Lwt.finalize
-        (fun () ->
-          let* block_count =
-            add_commits repo config.ncommits_trace commit_seq on_commit on_end
-              stats check_hash config.empty_blobs
-          in
-          [%logs.app "Closing repo..."];
-          let+ () = Store.Repo.close repo in
-          Stat_collector.close stats;
-          if not config.no_summary then (
-            [%logs.app "Computing summary..."];
-            Some (Trace_stat_summary.summarise ~block_count stat_path))
-          else None)
-        (fun () ->
-          if config.keep_stat_trace then (
-            [%logs.app "Stat trace kept at %s" stat_path];
-            Unix.chmod stat_path 0o444;
-            Lwt.return_unit)
-          else Lwt.return (Stat_collector.remove stats))
-    in
-    match summary_opt with
-    | Some summary ->
-        let p = Filename.concat config.artefacts_dir "stat_summary.json" in
-        Trace_stat_summary.save_to_json summary p;
-        fun ppf ->
-          Format.fprintf ppf "\n%t\n%a" repo_pp
-            (Trace_stat_summary_pp.pp 5)
-            ([ "" ], [ summary ])
-    | None -> fun ppf -> Format.fprintf ppf "\n%t\n" repo_pp
+    Lwt.finalize
+      (fun () ->
+        let* block_count =
+          add_commits repo config.number_of_commits_to_replay commit_seq
+            on_commit on_end stats check_hash config.empty_blobs
+        in
+        [%logs.app "Closing repo..."];
+        let+ () = Store.Repo.close repo in
+        Stat_collector.close stats;
+        [%logs.app "Computing summary..."];
+        Trace_stat_summary.summarise ~block_count stat_path)
+      (fun () ->
```

Comment:
Shouldn't we keep the `no-summary` here?

Reply:
I've made the replay always produce a summary as it simplifies a lot its UI. 

I now realize that it may be an annoying loss of time for when we replay lots of commits only for the stats trace. I will add something back

Reply:
Indeed, it's a good idea to save time.